### PR TITLE
Ansi color theming

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
This PR updates how the ansi-to-html script applies color codes so that themes can specify terminal colors.

For now I've just updated the dark theme:

Current:
image
New:
image

### CODE REVIEW

**Comment by jeremy-rifkin** [comment]
I'm not 100% happy with the color choices yet, I tried a number of themes from https://gogh-co.github.io/Gogh/ and https://glitchbone.github.io/vscode-base16-term/

**Comment by RubenRBS** [comment]
Ohhhhh looks super nice! Thanks a lot! :) I will take a proper look at this tomorrow 🫶

**Comment by mattgodbolt** [comment]
Thanks both! :) @RubenRBS is our resident colour expert :)

File: static/ansi-to-html.ts [hunk 1]
```diff
@@ -43,22 +43,22 @@ const defaults: AnsiToHtmlOptions = {
...
-        7: '#AAA',
-        8: '#555',
-        9: '#F55',
-        10: '#5F5',
-        11: '#FF5',
-        12: '#55F',
-        13: '#F5F',
-        14: '#5FF',
-        15: '#FFF',
+        0: 'var(--terminal-black)',
```
Comment (mattgodbolt) [inline static/ansi-to-html.ts:46]: Now I understand the question in Discord :)

File: static/styles/themes/dark-theme.scss [hunk 2]
```diff
@@ -681,3 +681,126 @@ textarea.form-control {
...
+    // --terminal-white: #FCFCFC;
+    // --terminal-bright-black: #7F8C8D;
+    // --terminal-bright-red: #C0392B;
+    // --terminal-bright-green: #1CDC9A;
+    // --terminal-bright-yellow: #FDBC4B;
+    // --terminal-bright-blue: #3DAEE9;
+    // --terminal-bright-magenta: #8E44AD;
+    // --terminal-bright-cyan: #16A085;
+    // --terminal-bright-white: #FFFFFF;
+    // Argonaut
```
Comment (mattgodbolt) [inline static/styles/themes/dark-theme.scss:755]: Not [thie one](https://en.wikipedia.org/wiki/Argonaut_Games) hehe.

File: static/styles/themes/dark-theme.scss [hunk 3]
```diff
@@ -681,3 +681,126 @@ textarea.form-control {
         }
     }
 }
+
+body {
+    // Terminal 3 and 4 bit color codes, colors come from https://gogh-co.github.io/Gogh/
```
Comment (mattgodbolt) [inline static/styles/themes/dark-theme.scss:686]: I'm not generally a fan of having large chunks of commented out code: mayyybe

```css
body.palette-dark-pastel {
 non-commented-out: #000;
...
}
```

or something? Or: maybe just leave the comment and link to the other possible choices?

But honestly it's also fine like this...

**Review by mattgodbolt — APPROVED** [review]
Looks good to me :)

File: static/styles/themes/dark-theme.scss (hunk 3 above)
Comment (jeremy-rifkin) [reply static/styles/themes/dark-theme.scss:686]: Thank you, I'll remove them when before this PR moves forward :) For now they're here for me to play with, and anyone else interested in taking a look. I've not found a color scheme I love yet.

File: static/ansi-to-html.ts (hunk 1 above)
Comment (jeremy-rifkin) [reply static/ansi-to-html.ts:46]: Seemed to be the way to go over having 32 class names :)

**Comment by partouf** [comment]
Is there any way this can be put under a toggle/setting? Colors from execution isn't super widely used (I think?), but I can imagine that if you do use colors on purpose, you might want them to be as intended.

**Comment by Endilll** [comment]
> Is there any way this can be put under a toggle/setting? Colors from execution isn't super widely used (I think?), but I can imagine that if you do use colors on purpose, you might want them to be as intended.

Now that Clang has merged https://github.com/llvm/llvm-project/pull/66514, ANSI colors will show up even more often, including problematic blue.

**Comment by jeremy-rifkin** [comment]
I think there is likely room for improvement with the colors but I think this is certainly an improvement over the previous coloring so I'd like to go ahead and merge this and iterate later.

It might be worthwhile for us to consider offering terminal color themes later on.

I think this is a reasonable color set for the moment:

image

**Pull request merged**